Source/Core/Core/State.cpp

```diff
@@ -354,6 +306,58 @@ static std::map<double, int> GetSavedStates()
   return m;
 }
 
+static void CompressBufferToFile(const u8* raw_buffer, u32 size, File::IOFile& f)
+{
+  u32 total_bytes_compressed = 0;
+
+  while (true)
+  {
+    u32 bytes_left_to_compress = size - total_bytes_compressed;
+
+    int bytes_to_compress =
+        static_cast<int>(std::min((u32)LZ4_MAX_INPUT_SIZE, bytes_left_to_compress));
+    int compressed_buffer_size = LZ4_compressBound(bytes_to_compress);
+    auto compressed_buffer = std::make_unique<char[]>(compressed_buffer_size);
+    int compressed_len =
+        LZ4_compress_default((char*)raw_buffer + total_bytes_compressed, compressed_buffer.get(),
+                             bytes_to_compress, compressed_buffer_size);
+
+    if (compressed_len == 0)
+      PanicAlertFmtT("Internal LZ4 Error - compression failed");
+
+    // The size of the data to write is 'compressed_len'
+    f.WriteArray(&compressed_len, 1);
```

Comment:
Although it's exceptionally unlikely that `int` is anything but 32 bits, it's probably still a good idea to ensure that since you're serializing it out.

Reply:
I assume changing `compressed_len` to a `int32_t` suffices?

Reply:
We usually use `s32` in Dolphin but yeah that's fine.